digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/GuestimateGeometryForMissingLinks.scala

```diff
@@ -100,4 +101,101 @@ class GuestimateGeometryForMissingLinks {
       }
     }
   }
+
```

Comment:
We could eliminate duplicate code and have a generic type with some match cases for different subtypes

Reply:
Good idea

Reply:
This looks like a good implementation.
![image](https://user-images.githubusercontent.com/13085707/32209956-bcda49dc-be13-11e7-92a2-01c5992136ea.png)